### DISCUSSION TITLE
[BUG] Get-PnpFile :  The -AsMemoryStream parameter has been revoked from m365PnP/Powershell docker container ?

### DISCUSSION THREAD

**Discussion by brotherSand** [pnp/powershell#3900] [General]
### Reporting an Issue or Missing Feature

I had a process I have been running for the last month where I pull files from SharePoint and load them to S3 using the m365pnp/powershell docker container.  The plan is to have this process run as a lambda.  So far it has been working with this being the method I was using to get the document contents: 

$streamData = (Get-PnPFile -Url $doc.ServerRelativeUrl -AsMemoryStream)

Yesterday I updated the docker container.  Now my process fails with this error: 

"Get-PnPFile: A parameter cannot be found that matches parameter name 'AsMemoryStream'."

The parameter is described very clearly here:  
https://pnp.github.io/powershell/cmdlets/Get-PnPFile.html

... and it was working.  What happened?  How did a useful method get subtracted?  

### What is the version of the Cmdlet module you are running?
ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Manifest   1.2.0                 PnP.PowerShell                      Core,Desk {Add-PnPClientSidePage, Add-PnPClientSidePageSection, Add-PnPClientSideText, Add-PnPClientSideWebPart…}

### Which operating system/environment are you running PnP PowerShell on?
- Docker container on Linux and AWS Lambda

**Comment by gautamdsheth** [comment]
hi @brotherSand , this parameter has not been removed. 
It is still there, you can check the source code here, must be something else that could have changed in your env :

https://github.com/pnp/powershell/blob/dev/src/Commands/Files/GetFile.cs

**Reply by brotherSand** [reply]
I rolled back to this version of the docker image and the code works again: 
m365pnp/powershell:2.4.16-nightly-alpine-3.14

But  m365pnp/powershell:latest   - This error occurs:
 "Get-PnPFile: A parameter cannot be found that matches parameter name 'AsMemoryStream'."

Same code, just different docker image. 

  Simple to replicate: 

========================================
```
$ docker run -it m365pnp/powershell
Unable to find image 'm365pnp/powershell:latest' locally
latest: Pulling from m365pnp/powershell
f7dab3ab2d6e: Already exists
f41cf633b820: Already exists
f746ece1c5fe: Already exists
df1656cddcd6: Pull complete
Digest: sha256:3bb46b90b2ba32455310da2fe9a0b80f69c448dbb3e737e0ce5fcb71887c14e4
Status: Downloaded newer image for m365pnp/powershell:latest
PowerShell 7.2.15
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS /> Get-PnPFile
Get-PnPFile         Get-PnPFileVersion

{checking with tab completion}

PS /> Get-PnPFile -
Url                           AsListItem                    AsFileObject                  Debug                         ErrorVariable                 OutBuffer
Path                          ThrowExceptionIfFileNotFound  Web                           ErrorAction                   WarningVariable               PipelineVariable
Filename                      AsString                      Connection                    WarningAction                 InformationVariable
AsFile                        Force                         Verbose                       InformationAction             OutVariable

PS /> Get-PnPFile -As
AsFile        AsListItem    AsString      AsFileObject
```

========================================

AsMemoryStream is not in tab-completion either.  

  Alternatively, with the older version ...

========================================
```
$ docker run -it m365pnp/powershell:2.4.16-nightly-alpine-3.14
PowerShell 7.2.15
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS /> Get-PnPFile -As
AsFile          AsListItem      AsString        AsFileObject    AsMemoryStream

```
========================================

It just appears to go away in the new version.
